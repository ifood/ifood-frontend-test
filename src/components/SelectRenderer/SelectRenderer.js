import React from 'react';
import PropTypes from 'prop-types';

import FormControl from '@material-ui/core/FormControl';
import InputLabel from '@material-ui/core/InputLabel';
import Select from '@material-ui/core/Select';
import MenuItem from '@material-ui/core/MenuItem';

const SelectRenderer = ({
  onChange,
  options,
  label,
  inputProps,
}) => ((
  <FormControl>
    <InputLabel htmlFor={inputProps.id}>{label}</InputLabel>
    <Select
      value=""
      onChange={onChange}
      inputProps={inputProps}
    >
      {
        options.map(option => ((
          <MenuItem
            key={option.value}
            value={option.value}
          >
            {option.name}
          </MenuItem>
        )))
      }
    </Select>
  </FormControl>
));

SelectRenderer.propTypes = {
  onChange: PropTypes.func,
  options: PropTypes.array,
  inputProps: PropTypes.object,
  label: PropTypes.string,
};

export default SelectRenderer;
