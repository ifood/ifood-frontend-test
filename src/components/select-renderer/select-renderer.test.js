import React from 'react';
import SelectRenderer from './select-renderer';
import { shallow } from 'enzyme';

// Mounting - Full DOM rendering including child components
// Shallow - Renders only the single component, not including its children,
// this is useful to isolate the component for pure unit testing.
// Render - Renders to static HTML, including children

it('should render correctly with props', () => {
    const fieldValues = {"country": "Brasil"};
    const name = "País";
    const values = [
        {
            "value": "AU",
            "name": "Australia"
        },
        {
            "value": "DE",
            "name": "Alemanhã"
        },
        {
            "value": "BR",
            "name": "Brasil"
        },
        {
            "value": "PT",
            "name": "Portugal"
        },
        {
            "value": "en_US",
            "name": "EUA"
        },
        {
            "value": "RU",
            "name": "Rússia"
        }
    ];

    const component = shallow(<SelectRenderer
        options={values}
        label={name}
        inputProps={{
          name: 'country',
          id: 'country',
        }}
        value={fieldValues}
        onChange={null}/>);
    expect(component).toMatchSnapshot();
});

