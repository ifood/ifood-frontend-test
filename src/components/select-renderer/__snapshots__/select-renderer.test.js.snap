// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly with props 1`] = `
<WithStyles(FormControl)>
  <WithStyles(WithFormControlContext(InputLabel))
    htmlFor="country"
  >
    País
  </WithStyles(WithFormControlContext(InputLabel))>
  <WithStyles(WithFormControlContext(Select))
    inputProps={
      Object {
        "id": "country",
        "name": "country",
      }
    }
    onChange={null}
    value={
      Object {
        "country": "Brasil",
      }
    }
  >
    <WithStyles(MenuItem)
      key="AU"
      value="AU"
    >
      Australia
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="DE"
      value="DE"
    >
      Alemanhã
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="BR"
      value="BR"
    >
      Brasil
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="PT"
      value="PT"
    >
      Portugal
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="en_US"
      value="en_US"
    >
      EUA
    </WithStyles(MenuItem)>
    <WithStyles(MenuItem)
      key="RU"
      value="RU"
    >
      Rússia
    </WithStyles(MenuItem)>
  </WithStyles(WithFormControlContext(Select))>
</WithStyles(FormControl)>
`;
